### PR TITLE
Fix Extra Links bug for project validators

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR fixes a bug introduced by https://github.com/code-dot-org/code-dot-org/pull/59283 which added project validator info/links to the Extra Links modal for lab2 labs. When a user with project validator permission is in start mode, e.g., '/levels/<level_id>/edit_blocks/start_sources' and then clicks on the 'Extra Links' button to open the modal, the page crashes since `project_info` and `owner_info` in `ExtraLinksModal.tsx` [are undefined](https://github.com/code-dot-org/code-dot-org/blob/438a3f4f2f34abf6fdb330ebbb1ddaf1a81232f2/apps/src/lab2/views/ExtraLinksModal.tsx#L339-L340).

Thanks to @mikeharv for reporting bug and @molly-moen for helping troubleshoot!

## Before update
Screenshot when project validator clicks on 'Extra Links' button in start mode:
Screenshot 2024-07-11 at 5 13 27 PM

## After Update
Screencast video:

https://github.com/user-attachments/assets/20e36777-b6f9-4ba4-bf6c-6076e5d4d1d9

## Links
[jira](https://codedotorg.atlassian.net/browse/LABS-916)
[Slack thread](https://codedotorg.slack.com/archives/C06JELCAPT9/p1720730438927709)

## Testing story

Tested in /allthethings for python lab levels. Interestingly, music lab does not have start mode.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lab2/views/ExtraLinksModal.tsx [hunk 1]
```diff
@@ -338,6 +338,9 @@ const ProjectLinkData: React.FunctionComponent<ProjectLinkDataProps> = ({
   }
   const ownerInfo = projectLinkData.owner_info;
   const projectInfo = projectLinkData.project_info;
+  if (!ownerInfo || !projectInfo) {
```
Comment (molly-moen) [inline apps/src/lab2/views/ExtraLinksModal.tsx:341]: can you also update the type of `projectLinkData` to indicate these could be undefined? And we could also have a string `message`?

File: dashboard/config/levels/custom/pythonlab/Allthethings Python Lab 1.level [hunk 2]
```diff
@@ -33,9 +32,9 @@
     "reference_links": [
       "/courses/csa-2023/guides/instantiating-objects"
     ],
-    "map_reference": "/docs/concepts/html/",
-    "teacher_markdown": "Teachers should see this extra info."
```
Comment (molly-moen) [inline dashboard/config/levels/custom/pythonlab/Allthethings Python Lab 1.level]: can you add this markdown back?

File: apps/src/lab2/views/ExtraLinksModal.tsx (hunk 1 above)
Comment (fisher-alice) [reply apps/src/lab2/views/ExtraLinksModal.tsx:341]: Yes -definitely. Will do - thanks!

**Review by molly-moen — APPROVED** [review]
Nice!

**Pull request merged**